tests/nat/builder/test_per_user_builder.py

```
@@ -0,0 +1,1225 @@
+class PerUserCounterWorkflowConfig(FunctionBaseConfig, name="per_user_counter_workflow"):
+    """A per-user workflow that uses the counter."""
+    counter_name: str = "counter"
+
+
+# Register all test components
+@pytest.fixture(scope="module", autouse=True)
+async def _register_components():
+
+    # Register shared function
+    @register_function(config_type=SharedFunctionConfig)
+    async def build_shared_fn(config: SharedFunctionConfig, b: Builder):
```

Ruff (0.14.8)

Unused function argument: `config`

(ARG001)